src/common/translate_item.js

```diff
@@ -105,8 +105,52 @@ Zotero.Translate.ItemSaver.prototype = {
 			payload.cookie = document.cookie;
 		}
 		payload.proxy = this._proxy && this._proxy.toJSON();
+		
+		// Add page data for snapshot
+		// TODO: Check for skipSnapshot
+		let singleFile;
+		for (let i = 0; i < items.length; i++) {
+			if (items[i].itemType !== 'webpage') {
```

Comment:
It doesn't need to be a 'webpage' item for there to be a snapshot. Any item type can have one. See, e.g., ACS Publications — the tests show a 'bookSection' item with a snapshot.